docs/azure.md

```diff
@@ -443,6 +443,37 @@ azure {
 }
 ```
 
+(azure-managed-identities)=
+
+## Managed identities
+
+:::{versionadded} 24.03.0-edge
+:::
+
+An Azure [managed identity](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview) can be used to authenticate with Azure Blob storage without using secrets such as a storage account key.
+
+The managed identity can be specified as follows:
+
+```groovy
+azure {
+    managedIdentity {
+        clientId = '<YOUR MANAGED IDENTITY>'
+        tenantId = '<YOUR TENANT ID>'
+    }
+
+    storage {
+        accountName = '<YOUR STORAGE ACCOUNT NAME>'
+    }
+
+    batch {
+        accountName = '<YOUR BATCH ACCOUNT NAME>'
+        location = '<YOUR BATCH ACCOUNT LOCATION>'
+    }
+}
+```
+
+Alternatively, you can set `azure.managedIdentity.system = true` to authenticate using the system-assigned identity.
```

Comment:
How is this expected to be used? likely it should be described the use and how to configure it

Reply:
@adamrtalbot can you add some details about how the system-assigned identity works? Ideally with a link to some Azure docs on the subject